### PR TITLE
fix(#861): Refactor wait action builder

### LINKED ISSUE #861: WaitConditionBuilder constructor never sets the condition in the builder despite taking it as an argument
**Citrus Version**
3.2.1

**Expected behavior**
The constructor should set the condition in the underlying builder.

**Actual behavior**
The constructor does not set the condition in the underlying builder, it has to be set manually.

**Test case sample**
```
Wait.Builder builder = new Wait.Builder()
        .time(timeout);
WaitActionConditionBuilder waitActionConditionBuilder = new WaitActionConditionBuilder(condition, builder);
```
does not work. The ```condition``` will never actually be set. Instead,
```
Wait.Builder builder = new Wait.Builder()
        .time(timeout)
        .condition(condition);

WaitActionConditionBuilder waitActionConditionBuilder = new WaitActionConditionBuilder(condition, waitBuilder);
```
has to be performed. This makes the argument ```condition``` in the ```WaitActionConditionBuilder``` useless.
I'd propose to add a line ```builder.condition(condition)``` in the constructor of ```WaitConditionBuilder```.

### PR DESCRIPTION
**Pull request by christophd**
- Improve constructors to only use builder instead of redundant condition and builder

Fixes #861

### CODE REVIEW

File: core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java [hunk 1]
```diff
@@ -25,26 +25,20 @@
...
     protected S self;
 
     /**
      * Default constructor using fields.
-     * @param condition
      * @param builder
      */
-    public WaitConditionBuilder(T condition, Wait.Builder builder) {
-        this.condition = condition;
+    public WaitConditionBuilder(Wait.Builder<T> builder) {
```
Comment (bbortt) [inline core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java:40]: the problem is, this change is potentially breaking @christophd.. should we target it for citrus 4.0? i am working on that, but have a silly error with `citrus-docker` and don't know what's up there.

Comment (bbortt) [reply core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java:40]: yes, it is never backwards compatible as soon as you just remove an argument from a constructor. if someone uses this constructor, he's doomed :(

Comment (bbortt) [reply core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java:40]: @christophd actually, I was just able to resolve the `citrus-docker` compilation failure.. maybe I can manage to provide a fullbuild of citrus@v4.0.0 until this evening.

Comment (christophd) [reply core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java:40]: @bbortt to be honest I think it would be ok to change the constructor even on a minor version upgrade. I know people may need to adjust their code but the change is trivial and the API change is done on a kind of internal API. People using the end user Java DSL will not even notice when updating. 

I know most of the libraries are doing it this way when the changes are trivial and when there is a notice of the change added to the release notes (that we do not have at the moment which is a problem).

Summing up I would add this for upcoming version 3.3.0 if we agree to have a decent release notes section.

Comment (bbortt) [reply core/citrus-base/src/main/java/com/consol/citrus/container/WaitConditionBuilder.java:40]: I am fine with that. did also note it in #878.

**Review by bbortt — APPROVED** [review]
